src/Hosting/Hosting/src/Internal/WebHostLifetime.cs

```diff
@@ -46,6 +52,12 @@ private void CancelKeyPress(object sender, ConsoleCancelEventArgs eventArgs)
         private void ProcessExit(object sender, EventArgs eventArgs)
         {
             Shutdown();
+            if (_exitedGracefully)
+            {
+                // On Linux if the shutdown is triggered by SIGTERM then that's signaled with the 143 exit code.
+                // Suppress that since we shut down gracefully. https://github.com/aspnet/AspNetCore/issues/6526
+                Environment.ExitCode = 0;
```

Comment:
How does this interact with a Program.Main() that returns a non-zero exit code?

Reply:
When the process is being terminated externally then Main isn't guaranteed to finish. There's a race.

Reply:
When the process isn't terminated externally, this sets the final exit code.
```cs
static int Main(string[] args)
{
    AppDomain.CurrentDomain.ProcessExit += delegate { Environment.ExitCode = 10; };
    return 5;
}
```
has an exit code of `10`.